# **Exercice 1**


L'analyse des données est l'ensemble des processus permettant de collecter, nettoyer, analyser, interpréter et restituer les données aux équipes métiers afin d'être un support d'aide à la prise de décision. Chaque étape de l'analyse comprend des processus différents :  
  
**La collecte des données** : Il s'agit de recueillir toutes les informations utiles pour mener à bien l'analyse. Ces informations peuvent provenir de différentes sources (sites web, rapports, fichiers, études de marché, etc). De plus ces données extraites peuvent être stockées dans plusieurs fichiers différents formats (csv, log, excel, image, texte etc)  
**Le nettoyage des données**: Lors de la collecte, les données sont recueillies de manière brute et de fois de manière désorganisée. Le processus de nettoyage consiste à, par exemple, supprimer les doublons ou encore extraire les données utiles à l'analyse.  
**L'analyse des données** : Ce processus consiste à vérifier la relation entre les variables explicatives, ou encore identifier les tendances.  
**L'interprétation des résultats** : Ici on tire des conlusions sur le pourquoi d'une action et on resout à ce problème.  
**La restitution** : Dans un projet d'analyse de données, plusieurs acteurs peuvent intervenir, après l'analyse et l'interprétation des données recueillies, les résultats sont transmis aux équipes métiers afin qu'ils puissent prendre des décisions.  
  
Dans l'ère moderne, ce sont des millions de données qui sont traitées chaque jour, l'exploration de ces données devient donc cruciale pour les prises de décision rapide. Cette analyse peut permettre les prises de décision rapide, permet d'avoir un avantage concurrentiel, de comprendre mieux les besoins du client et de gérer les risques.  
  
L'analyse de données peut être utilisée dans plusieurs secteurs, la banque avec la detection de fraute, la sécurité routière avec la détection des zones à risque etc

# **Exercice 2**

In [1]:
import pandas as pd

In [16]:
df1 = pd.read_csv('clean_dataset.csv')
df1.head(5)
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Gender          690 non-null    int64  
 1   Age             690 non-null    float64
 2   Debt            690 non-null    float64
 3   Married         690 non-null    int64  
 4   BankCustomer    690 non-null    int64  
 5   Industry        690 non-null    object 
 6   Ethnicity       690 non-null    object 
 7   YearsEmployed   690 non-null    float64
 8   PriorDefault    690 non-null    int64  
 9   Employed        690 non-null    int64  
 10  CreditScore     690 non-null    int64  
 11  DriversLicense  690 non-null    int64  
 12  Citizen         690 non-null    object 
 13  ZipCode         690 non-null    int64  
 14  Income          690 non-null    int64  
 15  Approved        690 non-null    int64  
dtypes: float64(3), int64(10), object(3)
memory usage: 86.4+ KB


Le fichier télécharger est un fichier csv de 690 lignes 16 colonnes dont le genre de la personne, l'age, sa situation matrimonial ou encore sa ville de résidence

In [17]:
df1 = pd.read_csv("Mental health Depression disorder Data.csv")
df1.head(5)
df1.columns

/tmp/ipykernel_6634/2997855578.py:1: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  df1 = pd.read_csv("Mental health Depression disorder Data.csv")


Index(['index', 'Entity', 'Code', 'Year', 'Schizophrenia (%)',
       'Bipolar disorder (%)', 'Eating disorders (%)', 'Anxiety disorders (%)',
       'Drug use disorders (%)', 'Depression (%)',
       'Alcohol use disorders (%)'],
      dtype='object')

Les données recueillies sont des données concernant la dépression mental des habitants de certains pays et comportes 108553 patients ainsi que 10 colonnes représentant le pays, le code, l'année, la dépression (%), etc

In [14]:
df2 = pd.read_csv('Time Americans Spend Sleeping.csv')
df2.head(5)
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    object 
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    object 
 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB


Le fichier charger présente le temps que les américains passent à dormir, il ppossède 945 échantillons et 9 valiable dont l'index, l'année, la période, le groupe d'age, etc.

## **Exercice 3 : Identification des types de données**

### **Rappel**

**Données qualitatives** : Ces données se concentrent sur les descriptions, les opinions, les émotions et les comportements. Elles permettent de comprendre le contexte, les motivations profondes ou les mécanismes derrière un phénomène. On peut avoir des informations textuelles (Non numériques), des questions posées, etc.      
**Données quantitatives**: Ces données se concentrent sur les chiffres, les statistiques et les mesures. Elles permettent de quantifier un phénomène et d'analyser des tendances sur de grands échantillons. On peut avoir des données numériques, et des questions posée ou peut chercher à savoir (Combien, À quelle fréquence, etc.) 

#### **Fichier clean_dataset.csv**
**Données** : Gender, Age, Debt, Married, BankCustomer, Industry, Ethnicity, YearsEmployed, PriorDefault, Employed, CreditScore, DriversLicense, Citizen, ZipCode, Income, Approved
  
**Données qualitatives** : Industry, Ethnicity, Citizen
**Données quantitatives** : Gender, Age, Debt, Married, BankCustomer, YearsEmployed, PriorDefault, Employed, CreditScore, DriversLicense, ZipCode, Income, Approved

#### **Mental health Depression disorder Data.csv**
**Données** : index, Entity, Code, Year, Schizophrenia (%), Bipolar disorder (%), Eating disorders (%), Anxiety disorders (%), Drug use disorders (%), Depression (%), Alcohol use disorders (%)
  
**Données qualitatives** : Entity, Code, Year, Schizophrenia (%), Bipolar disorder (%), Eating disorders (%)
**Données quantitatives** : Anxiety disorders (%), Drug use disorders (%), Depression (%), Alcohol use disorders (%)

#### **Time Americans Spend Sleeping.csv**
**Données** : index, Year, Period,  Avg hrs per day sleeping, Standard Error, Type of Days, Age Group, Activity, Sex 
  
**Données qualitatives** : Period, Type of Days, Age Group, Activity, Sex  
**Données quantitatives** : index, Year, Avg hrs per day sleeping, Standard Error,

##  **Exercice 4**

In [18]:
df3 = pd.read_csv('Iris.csv')
df3

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa
...,...,...,...,...,...,...
145,146,6.7,3.0,5.2,2.3,Iris-virginica
146,147,6.3,2.5,5.0,1.9,Iris-virginica
147,148,6.5,3.0,5.2,2.0,Iris-virginica
148,149,6.2,3.4,5.4,2.3,Iris-virginica


Données : Id, SepalLengthCm, SepalWidthCm, PetalLengthCm, PetalWidthCm, Species

Nous avons dit précendement que les données qualitative étaient des données principalement textuelles et les données quantitatives étaient des données principalement numériques.
Dans notre dataset, les colonnes peuvent avoir différent types (object pour représenter les variables qualitatives et int, float pour représenter les donnnées quantitatives).  
En faisant df3.info(), on obtient des informations sur le type de chaque colonne 

In [19]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


On observe qu'il n'y a qu'une seule variable qualitative (Species) et cinq variables quantitatives (Id, SepalLengthCm, SepalWidthCm, PetalLengthCm, PetalWidthCm)

## **Exercice 5**

In [20]:
df4 = df2

In [21]:
df4.head()

,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


In [26]:
df4['Sex'].unique()

array(['Both', 'Men', 'Women'], dtype=object)

Ici, on peut d'abord ciblé le temps moyen de sommeil d'un américain ('Avg hrs per day sleeping'), on verra s'il dort beaucoup ou pas, ensuite, on peut vérifier quelle catégorie d'âge est le plus impliqué (enfant, jeune, adulte), on pourra ainsi voir la catégorie d'âge sur laquelle va se porter notre analyse, à cela on ajouter le sex de la personne concernée pour voir entre homme et femme, qui est le plus impactés  
On peut aussi grouper ces variables pour voir, par exemple si, le temps moyen de sommeil dépend de la catégorie d'âge ou du Sex

## **Exercice 6**

**Les rapports financiers d'une entreprise sont stockés dans un fichier Excel** : Fichier structurées  
**Photographies téléchargées sur une plateforme de médias sociaux** : Fichier non structuré  
**Une collection d'articles de presse sur un site web** : Sémi-structuré  
**Données d'inventaire dans une base de données relationnelle** : Fichier structuré  
**Enregistrements d'entretiens issus d'une étude de marché** : Fichier non struturé  

## **Exercice 7**

**Une série d'articles de blog sur les expériences de voyage** : On peut uiliser Traitement Automatique du Langage Naturel (NLP) et le text mining pour extraire automatiquement le titre, le pays, la ville, la date, les sentiment, l'activités.  
Les articles de blog sont des textes libres. Le NLP permet d’identifier automatiquement des éléments utiles (lieux, dates, émotions, activités) afin de les stocker dans une base de données exploitable pour l’analyse ou la recommandation.  
**Enregistrements audio des appels au service client** : Utilisation de la reconnaissance vocale (Speech-to-Text) puis appliquer du NLP pour convertir l’audio en texte puis structurer les données en fonction du client, du problème, de la durée, du type de résolution.  
Les données audio ne sont pas directement exploitables par les systèmes analytiques. La transcription transforme la voix en texte, puis les outils NLP permettent d’extraire des informations utiles pour améliorer le service client.  
**Notes manuscrites issues d'une séance de brainstorming** : Utilisation de l’OCR (Reconnaissance Optique de Caractères) associé à l’analyse de texte pour transformer les notes manuscrites en texte numérique structuré, on peut receuillir l'idée principale, la catégorie, les priorités.  
Les notes manuscrites sont difficiles à exploiter automatiquement. L’OCR permet de convertir l’écriture en texte numérique, ensuite organisé en catégories pour faciliter le suivi des idées et la gestion de projet.  
**Un tutoriel vidéo sur la cuisine**: Utilisation de la vision par ordinateur et la transcription audio pour extraire les étapes, les ingrédients, l'action, le temps mis.  
Une vidéo contient plusieurs types de données (image, son, texte). Combiner vision par ordinateur et transcription audio permet d’identifier les ingrédients, les actions et les étapes afin de créer une recette structurée et consultable facilement.

## **Exercice 8**

In [27]:
df5 = pd.read_csv('train.csv')

In [28]:
df5.head(5)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## **Exercice 9**

In [30]:
data = {
    'Name': ['John', 'Anna', 'Peter', 'Linda'],
    'Age': [28, 34, 29, 32],
    'City': ['New York', 'Paris', 'Berlin', 'London']
}
df6 = pd.DataFrame(data)

In [31]:
df6

,Name,Age,City
0,John,28,New York
1,Anna,34,Paris
2,Peter,29,Berlin
3,Linda,32,London


In [34]:
df6.to_excel('user.xlsx',index=False)

In [35]:
df6.to_json('user.json', index= False)

## **Exercice 10**

In [36]:
df7 = pd.read_json('posts.json')

In [38]:
df7.head(5)

,userId,id,title,body
0,1,1,sunt aut facere repellat provident occaecati e...,quia et suscipit\nsuscipit recusandae consequu...
1,1,2,qui est esse,est rerum tempore vitae\nsequi sint nihil repr...
2,1,3,ea molestias quasi exercitationem repellat qui...,et iusto sed quo iure\nvoluptatem occaecati om...
3,1,4,eum et est occaecati,ullam et saepe reiciendis voluptatem adipisci\...
4,1,5,nesciunt quas odio,repudiandae veniam quaerat sunt sed\nalias aut...
